src/content/guides/shimming.md

```diff
@@ -13,233 +13,434 @@ related:
+Let's move our `import` to a new file and add the [`whatwg-fetch`](https://github.com/github/fetch) polyfill:
 
+``` bash
+npm i --save whatwg-fetch
+```
 
-## Loading polyfills on demand
+__src/index.js__
 
-It's common in web projects to include polyfills in the main bundle. This is not recommended because we are penalizing modern browsers users by making them download a bigger file with unneeded scripts.
+``` diff
+- import 'babel-polyfill';
+-
+  function component() {
+    var element = document.createElement('div');
 
-The simplest way to mitigate this is by adding a separate entry point in your webpack config file including the polyfills your project needs.
+    element.innerHTML = join(['Hello', 'webpack'], ' ');
 
-```javascript
-// webpack.config.js
-module.exports = {
-  entry: {
-    polyfills: [
-      'babel-polyfill',
-      'whatwg-fetch'
-    ],
-    main: './src/index.js'
+    return element;
   }
-  // ... rest of your webpack config
-};
+
+  document.body.appendChild(component());
 ```
 
-An alternative is to create a new entry file and manually import these packages.
+__project__
+
+``` diff
+  webpack-demo
+  |- package.json
+  |- webpack.config.js
+  |- /dist
+  |- /src
+    |- index.js
+    |- globals.js
++   |- polyfills.js
+  |- /node_modules
+```
+
+__src/polyfills.js__
 
 ```javascript
-// src/polyfills.js
 import 'babel-polyfill';
 import 'whatwg-fetch';
 ```
 
-```javascript
-// webpack.config.js
-module.exports = {
-  entry: {
-    polyfills: './src/polyfills.js',
-    main: './src/index.js'
-  }
-  // rest of your webpack config
-};
+__webpack.config.js__
+
+``` diff
+  const path = require('path');
+
+  module.exports = {
+-   entry: './src/index.js',
++   entry: {
++     polyfills: './src/polyfills.js',
++     index: './src/index.js'
++   },
+    output: {
+-     filename: 'bundle.js',
++     filename: '[name].bundle.js',
+      path: path.resolve(__dirname, 'dist')
+    },
+    module: {
+      rules: [
+        {
+          test: require.resolve('index.js'),
+          use: 'imports-loader?this=>window'
+        },
+        {
+          test: require.resolve('globals.js'),
+          use: 'exports-loader?file,parse=helpers.parse'
+        }
+      ]
+    },
+    plugins: [
+      new webpack.ProvidePlugin({
+        join: ['lodash', 'join']
+      })
+    ]
+  };
 ```
 
-In your html file you need to conditionally load the `polyfills.js` file before your bundle. How you make this decision depends on the technologies and browsers you need to support.
+With that in place, we can add the logic to conditionally load our new `polyfills.bundle.js` file. How you make this decision depends on the technologies and browsers you need to support. We'll just do some simple testing to determine whether our polyfills are needed:
+
```

Comment:
See the webpack config change: `filename: '[name].bundle.js',`.

It would be `polyfills.bundle.js`. I think it's better that way as it's clear that the `polyfills.js` went through webpack instead of being loaded directly. Let me know if I'm missing something here.